src/tests/unit/test_tree_functionality/test_tree_node_delete.py

```diff
@@ -0,0 +1,167 @@
+import unittest
```

Comment:
all tests are copied from the old "test_tree_node_helpers.py" file without being modified.